lib/Widgets/TransmissionListEntry.vala

```diff
@@ -45,36 +48,44 @@ public class Drop.Widgets.TransmissionListEntry : Gtk.Grid {
     }
 
     private void build_ui () {
-        this.column_spacing = 6;
-        this.row_spacing = 6;
+        this.orientation = Gtk.Orientation.HORIZONTAL;
+        this.spacing = 12;
+
+        main_grid = new Gtk.Grid ();
+        main_grid.column_spacing = 12;
+        main_grid.row_spacing = 2;
 
         icon = new Gtk.Image ();
 
         primary_label = new Gtk.Label ("");
         primary_label.halign = Gtk.Align.START;
+        primary_label.valign = Gtk.Align.END;
         primary_label.ellipsize = Pango.EllipsizeMode.MIDDLE;
-        primary_label.get_style_context ().add_class (Granite.StyleClass.H3_TEXT);
+        primary_label.use_markup = true;
 
         secondary_label = new Gtk.Label ("");
         secondary_label.halign = Gtk.Align.START;
+        secondary_label.valign = Gtk.Align.START;
         secondary_label.hexpand = true;
 
         progress_revealer = new Gtk.Revealer ();
-        progress_revealer.reveal_child = false;
         progress_revealer.transition_type = Gtk.RevealerTransitionType.SLIDE_DOWN;
 
         progress_bar = new Gtk.ProgressBar ();
+        progress_bar.valign = Gtk.Align.CENTER;
 
         progress_revealer.add (progress_bar);
 
-        action_area = new Gtk.Box (Gtk.Orientation.VERTICAL, 0);
-        action_area.valign = Gtk.Align.END;
+        action_area = new Gtk.Box (Gtk.Orientation.VERTICAL, 4);
```

Comment:
Use 3, 6 or 12. The elementary designers hate numbers that are not devidable with 3. :smile: